cdap-examples/SpamClassifier/src/main/java/co/cask/cdap/examples/sparkstreaming/SpamClassifier.java

```diff
@@ -0,0 +1,112 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.examples.sparkstreaming;
+
+import co.cask.cdap.api.Resources;
+import co.cask.cdap.api.annotation.UseDataSet;
+import co.cask.cdap.api.app.AbstractApplication;
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.data.schema.UnsupportedTypeException;
+import co.cask.cdap.api.data.stream.Stream;
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.lib.ObjectStore;
+import co.cask.cdap.api.dataset.lib.ObjectStores;
+import co.cask.cdap.api.service.Service;
+import co.cask.cdap.api.service.http.AbstractHttpServiceHandler;
+import co.cask.cdap.api.service.http.HttpServiceRequest;
+import co.cask.cdap.api.service.http.HttpServiceResponder;
+import co.cask.cdap.api.spark.AbstractSpark;
+import com.google.common.base.Charsets;
+
+import java.net.HttpURLConnection;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+/**
+ * Spark Streaming Example for classifying kafka messages through Spark MlLib NaiveBayesModel
+ */
+public class SpamClassifier extends AbstractApplication {
+
+  static final String SERVICE_HANDLER = "MessageClassification";
+  public static final String STREAM = "trainingDataStream";
+  public static final String DATASET = "messageClassificationStore";
+
+  @Override
+  public void configure() {
+    setName("SpamClassifier");
+    setDescription("A Spark Streaming Example for Kafka Message Classification");
+    addStream(new Stream(STREAM));
+    addSpark(new SparkSpamFilteringSpecification());
+    addService(SERVICE_HANDLER, new SpamClassifierServiceHandler());
+
+    // Store for message classification status
+    try {
+      ObjectStores.createObjectStore(getConfigurer(), DATASET, Double.class,
+                                     DatasetProperties.builder().setDescription("Kafka Message Spam " +
+                                                                                  "Classification").build());
+    } catch (UnsupportedTypeException e) {
+      // This exception is thrown by ObjectStore if its parameter type cannot be
+      // (de)serialized (for example, if it is an interface and not a class, then there is
+      // no auto-magic way deserialize an object.) In this case that will not happen
+      // because Double is actual classes.
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * A Spark Streaming program which classifies kafka messages as spam or ham
+   */
+  public static final class SparkSpamFilteringSpecification extends AbstractSpark {
```

Comment:
You actually don't need this. Better merge this with the `SpamClassifierProgram`.